index.js

```diff
@@ -53,7 +56,13 @@ Taplytics.newAsyncVariable = (name, defaultValue, callback) => {
 
   let args = [name, defaultValue]
   if (Platform.OS === 'ios') {
-    args.push(cb)
+    Taplytics.nativeEventEmitter.addListener('asyncVariable', ({id, value}) => {
+      const cb = asyncVarCallbackDispatcher.get(id)
+      if (cb) cb(null, value)
+    })
+    args.push(asyncVarCallbackId)
+    asyncVarCallbackDispatcher.set(asyncVarCallbackId, cb)
+    asyncVarCallbackId++
```

Comment:
it's possible there could be a race condition here where you can set two different callbacks to 0 and the second one will override the first one

Reply:
Aren't synchronous functions in JS run all together? Like how JS only runs on a single thread?